src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -1,17 +1,266 @@
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
+                        self._iter_luminescence_measurement_document(
+                            plate_block, position
+                        ),
+                    ),
+                )
+            )
+        elif plate_block.plate_block_type == "Fluorescence":
+            return PlateReaderDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
+                        self._iter_fluorescence_measurement_document(
+                            plate_block, position
+                        )
+                    ),
+                )
+            )
+        else:
+            error = "Unable to find valid plate block type."
+            raise AllotropeConversionError(error)
+
+    def _iter_fluorescence_measurement_document(
+        self, plate_block: PlateBlock, position: str
+    ) -> Iterator[FluorescencePointDetectionMeasurementDocumentItems]:
+        if plate_block.header.scan_position == "TRUE":
+            scan_position = (
+                ScanPositionSettingPlateReader.bottom_scan_position__plate_reader_
+            )
+        elif plate_block.header.scan_position == "FALSE":
+            scan_position = (
+                ScanPositionSettingPlateReader.top_scan_position__plate_reader_
+            )
+        else:
+            error = "Unable to find valid scan position."
+            raise AllotropeConversionError(error)
+
+        reads_per_well = assert_not_none(
+            plate_block.header.reads_per_well,
+            msg="Unable to find plate block reads per well.",
+        )
+
+        for idx, data_element in enumerate(
+            plate_block.block_data.iter_wavelengths(position)
+        ):
+            yield FluorescencePointDetectionMeasurementDocumentItems(
+                measurement_identifier=str(uuid.uuid4()),
+                fluorescence=TRelativeFluorescenceUnit(value=data_element.value),
+                compartment_temperature=(
+                    None
+                    if data_element.temperature is None
+                    else TQuantityValueDegreeCelsius(data_element.temperature)
+                ),
+                sample_document=SampleDocument(
+                    location_identifier=data_element.position,
+                    well_plate_identifier=plate_block.header.name,
+                    sample_identifier=f"{plate_block.header.name} {data_element.position}",
```

Comment:
general comment for this and following methods: `plate_block.header` is used a lot. consider extracting local vars for readability.